src/main/kotlin/com/pubnub/api/crypto/cryptor/LegacyCryptor.kt

```diff
@@ -0,0 +1,99 @@
+package com.pubnub.api.crypto.cryptor
+
+import com.pubnub.api.PubNubError
+import com.pubnub.api.PubNubException
+import com.pubnub.api.crypto.data.EncryptedData
+import com.pubnub.api.crypto.data.EncryptedStreamData
+import java.io.InputStream
+import java.security.MessageDigest
+import java.security.spec.AlgorithmParameterSpec
+import java.util.Random
+import javax.crypto.Cipher
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.SecretKeySpec
+
+private const val STATIC_IV = "0123456789012345"
+private const val CIPHER_TRANSFORMATION = "AES/CBC/PKCS5Padding"
+
+class LegacyCryptor(val cipherKey: String, val useRandomIv: Boolean = true) : Cryptor {
+
+    override fun id(): ByteArray {
+        return ByteArray(4) { 0.toByte() } // it was agreed that legacy PN Cryptor will have 0 as ID
+    }
+
+    // for this Cryptor metadata stored in EncryptedData contains random or static iv
+    override fun encrypt(data: ByteArray): EncryptedData {
+        return try {
+            val ivBytes: ByteArray = getIvBytesForEncryption()
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val keyBytes: ByteArray = getKeyAsShaInHexBytes(cipherKey)
+            val newKey: SecretKeySpec = SecretKeySpec(keyBytes, "AES")
+            val cipher: Cipher = Cipher.getInstance(CIPHER_TRANSFORMATION)
+            cipher.init(Cipher.ENCRYPT_MODE, newKey, ivSpec)
+            val encrypted: ByteArray = cipher.doFinal(data)
+            val encryptedWithIV: ByteArray = ivBytes + encrypted
+
+            EncryptedData(
+                encryptedWithIV,
+                ivBytes
+            ) // todo legacy encrypt method returns String that is Base64. How to handle this?
+        } catch (e: Exception) {
+            throw PubNubException(errorMessage = e.message, pubnubError = PubNubError.CRYPTO_ERROR)
+        }
+    }
+
+    override fun decrypt(encryptedData: EncryptedData): ByteArray {
+        return try {
+            val ivBytes: ByteArray = getIvBytesForDecryption(encryptedData)
+            val ivSpec: AlgorithmParameterSpec = IvParameterSpec(ivBytes)
+            val keyBytes: ByteArray = getKeyAsShaInHexBytes(cipherKey)
+            val newKey = SecretKeySpec(keyBytes, "AES")
```

Comment:
Should this be rather part of setting up the cryptor? Why do those operations every time we call it. Some of them can be did just once